File: level1/python/로또의 최고 순위와 최저 순위 77484.py
def solution(lottos, win_nums):
    lank = {6:1, 5:2, 4:3, 3:4, 2:5, 1:6, 0:6} # 일치한(맞힌) 번호 갯수에 따른 순위 정보
    zero_cnt = lottos.count(0)
   
    min = len(set(lottos) & set(win_nums)) # 확실히 맞힌 원소의 개수 : 최저 순위
    max = min + zero_cnt # 최대한으로 맞혔다고 가정했을 때, 맞힌 원소의 개수 : 최고 순위

    return [lank[max], lank[min]] # 최고 순위, 최저 순위
print(solution([45, 4, 35, 20, 3, 9],[20, 9, 3, 45, 4, 35]))

File: level1/javascript/로또의 최고 순위와 최저 순위 77484.js
function solution(lottos, win_nums) {
    const lank = {6:1, 5:2, 4:3, 3:4, 2:5, 1:6, 0:6}; // 맞힌 번호 개수에 따른 순위 정보
    let low = lottos.filter(x => win_nums.includes(x)).length; // 최저 순위 : 가장 적게 맞히는 경우, 정답 배열과 완전 일치하는 원소만 맞혔다고 간주
    let high = lottos.filter(x => x == 0).length + low; // 최고 순위 : 가장 많이 맞히는 경우, 알아볼 수 없는 번호(0)도 모두 맞힌 것으로 간주
    
    return [lank[high], lank[low]];
}

// 테스트 코드
console.log(solution([44, 1, 0, 0, 31, 25],[31, 10, 45, 1, 6, 19]));